buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/YamlRestCompatTestPlugin.java

```diff
@@ -73,9 +73,11 @@ public void apply(Project project) {
         SourceSet yamlTestSourceSet = sourceSets.getByName(YamlRestTestPlugin.SOURCE_SET_NAME);
         GradleUtils.extendSourceSet(project, YamlRestTestPlugin.SOURCE_SET_NAME, SOURCE_SET_NAME);
 
-        // create the test cluster container, and always use the default distribution
+        // create the test cluster container, and use the default distribution. System property is only used to help test this plugin.
```

Comment:
Wasn't a fan of doing this...but for reasons I have hard time following, this requires the `:distribution:archives:linux-tar` (my workstation is a linux box) project to exist. I assume that OS specific, and did not want to have to mirror all of the variants from distribution:archive as dummy projects. 

Alternatively, I think i can only run this test specifically for a given OS, but avoided that since there is some file copying going on and wanted to get coverage across OS's. The distribution in use is not an important detail of the test.

Reply:
I mentioned in another comment that you should be able to just configure the distribution explicitly in the tests so this sys property in the production code should not be required. IMO its fine to not test all OS specific resolutions as this is mostly logic done in the distribution-download-plugin and the test cluster plugin which we test separately